### PR TITLE
day2

### PR DESCRIPTION
**Pull request by ShayRasooly**

### CODE REVIEW

File: Week1/Day2/DailyChallenge/script.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Challenge 1
+inputted_number = int(input('Please give a number: '))
+length = int(input('Please give a length: '))
+
+list_of_multiples = []
+for number in range(length):
+  list_of_multiples.append(inputted_number * (number + 1))
```
Comment (ShayRasooly) [inline Week1/Day2/DailyChallenge/script.py:7]: maybe a bit more clear names than "number" and "inputted number"?

File: Week1/Day2/DailyChallenge/script.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Challenge 1
+inputted_number = int(input('Please give a number: '))
+length = int(input('Please give a length: '))
+
+list_of_multiples = []
+for number in range(length):
```
Comment (ShayRasooly) [inline Week1/Day2/DailyChallenge/script.py:6]: maybe you can change the range instead of +1?

File: Week1/Day2/DailyChallenge/script.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  list_of_multiples.append(inputted_number * (number + 1))
+print(list_of_multiples)
+
+# Challenge 2
+inputted_string = input("Please enter a word: ")
+
+new_string = inputted_string[0]
+for position in range(len(inputted_string)):
+  if position == 0:
```
Comment (ShayRasooly) [inline Week1/Day2/DailyChallenge/script.py:15]: maybe there is a way to get rid of this condition?

File: Week1/Day2/DailyChallenge/script.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    continue
+  elif inputted_string[position] != new_string[-1]:
```
Comment (ShayRasooly) [inline Week1/Day2/DailyChallenge/script.py:17]: a bit weird comparison

File: Week1/Day2/ExercisesXP/script.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+my_fav_numbers.add(56)
+my_fav_numbers.add(84)
+print(my_fav_numbers)
+
+my_fav_numbers.discard(84)
+print(my_fav_numbers)
+
+friend_fav_numbers = {5, 9, 78, 81}
+our_fav_numbers = friend_fav_numbers
+for number in my_fav_numbers:
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:12]: maybe there is a better way of doing this?

File: Week1/Day2/ExercisesXP/script.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+number_of_apple_trays = basket.count("Apples")
+basket.clear()
+print(basket)
+
+# Exercise 4
+# 1. A float is a number that contains decimals
+# 2.
+list_of_floats = []
+for integer in range(3, 11):
+  half_of_integer = integer / 2
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:34]: in a different language this wouldn't work.. can you tell me why and how to fix it?

File: Week1/Day2/ExercisesXP/script.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+while number <= 4.5:
+  number += 0.5
+  if number == int(number):
+    number = int(number)
+  list_of_floats_2.append(number)
+print(list_of_floats_2)
+
+# Exercise 5
+# 1.
+for number in range(20):
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:52]: maybe look again at the range video, range can take more than 1 argument

File: Week1/Day2/ExercisesXP/script.py [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+  print(number + 1)
+
+# 2.
+for number in range(20):
+  if (number + 1) % 2 == 0:
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:57]: what is the difference between this number and its index?

File: Week1/Day2/ExercisesXP/script.py [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+    print(number + 1)
+
+# Exercise 6
+my_name = "David"
+username = "Whatever"
+while username != my_name:
+  username = input("Please enter your name starting with a capital letter: ")
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:64]: is there a way to not bother the user with this request?

File: Week1/Day2/ExercisesXP/script.py [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:65]: what about some messages so the user can know if he guessed correct or not

File: Week1/Day2/ExercisesXP/script.py [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+  print("You chose one of your favorite fruits")
+else:
+  print("You chose a new fruit. I hope you enjoy.")
+
+# Exercise 8
+pizza_toppings = []
+while True:
+  pizza_topping = input("Please enter the pizza topping you wanna add or 'quit' to move on with your order: ")
+  if pizza_topping == 'quit':
+    break
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:81]: is there a way to avoid this break?

File: Week1/Day2/ExercisesXP/script.py [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+print(f"You'll get {pizza_toppings} for a total price of {total_price} euro.")
+
+# Exercise 9
+# 1. 2. 3.
+family = []
+total_cost = 0
+while True:
+  age = input("Please enter the age of the next person who wants to watch the movie or 'quit': ")
+  if age == 'quit':
+    break
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:95]: again. i advise against using break

File: Week1/Day2/ExercisesXP/script.py [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+    total_cost += 15
+print(f'Your total cost is {total_cost} euro.')
+
+# 4.
+names = ["David", "Marija", "Manon"]
+def age_checker(list_of_names):
+  names_for_removal = []
+  for person in list_of_names:
+    age = input(f'Dear {person}, please enter your age: ')
+    if int(age) < 21: # Assuming that if restricted for 16 to 21 it is also restricted for under 16
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:111]: nice assumption.

File: Week1/Day2/ExercisesXP/script.py [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+      names_for_removal.append(person)
+  for person in names_for_removal:
+    names.remove(person)
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:114]: is there more efficient way to remove a list from another list?

File: Week1/Day2/ExercisesXP/script.py [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+
+age_checker(names)
+
+# Exercise 10
+sandwich_orders = ["Tuna sandwich", "Pastrami sandwich", "Avocado sandwich", "Pastrami sandwich", "Egg sandwich", "Chicken sandwich", "Pastrami sandwich"]
+
+order_number = 0
+while order_number < len(sandwich_orders):
+  if "Pastrami" in sandwich_orders[order_number]:
+    sandwich_orders.remove(sandwich_orders[order_number]) # after removing it is not necessary to add 1 to order_number
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:125]: you're sure this is how "remove" works?

File: Week1/Day2/ExercisesXP/script.py [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+  else:
+    order_number += 1
+print(sandwich_orders)
+
+finished_sandwiches = []
+while len(sandwich_orders) > 0:
+   sandwich_in_preparation = sandwich_orders.pop(0)
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:132]: nice name and pop

File: Week1/Day2/ExercisesXP/script.py [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+   finished_sandwich = sandwich_in_preparation.lower()
+   finished_sandwiches.append(finished_sandwich)
+
+for sandwich in finished_sandwiches:
```
Comment (ShayRasooly) [inline Week1/Day2/ExercisesXP/script.py:136]: maybe there is a way to avoid this loop?

File: Week1/Day2/DailyChallenge/script.py (hunk 1 above)
Comment (David-7805) [reply Week1/Day2/DailyChallenge/script.py:7]: Maybe:
for factor in range(1, length + 1):
----list_of_multiples.append(base_number * factor) ()

Assuming that earlier I stored the input into a variable called base_number.

File: Week1/Day2/DailyChallenge/script.py (hunk 2 above)
Comment (David-7805) [reply Week1/Day2/DailyChallenge/script.py:6]: Yes, you are right!
for factor in range(1, length + 1):
----list_of_multiples.append(base_number * factor)

File: Week1/Day2/DailyChallenge/script.py (hunk 3 above)
Comment (David-7805) [reply Week1/Day2/DailyChallenge/script.py:15]: Yes, you are right!

for position in range(1, len(inputted_string)):

File: Week1/Day2/DailyChallenge/script.py (hunk 4 above)
Comment (David-7805) [reply Week1/Day2/DailyChallenge/script.py:17]: Why? Is it that bad?

File: Week1/Day2/ExercisesXP/script.py (hunk 5 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:12]: You are right!

our_fav_numbers = my_fav_numbers.union(friend_fav_numbers)

File: Week1/Day2/ExercisesXP/script.py (hunk 13 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:111]: Yes, I had to assume something here : )

File: Week1/Day2/ExercisesXP/script.py (hunk 15 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:125]: Yes, I think it works like that.
test_list = [5, 7, 9, 14, 28]
test_list.remove(test_list[3])
print(test_list)
[5, 7, 9, 28]
The next element in the list is at the same index as the element that was removed before.

File: Week1/Day2/ExercisesXP/script.py (hunk 16 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:132]: Thank you, Shay : )

File: Week1/Day2/ExercisesXP/script.py (hunk 7 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:52]: for number in range(1, 21):

File: Week1/Day2/ExercisesXP/script.py (hunk 6 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:34]: Maybe in another language a division of 2 integers returns only an integer. Therefore:
half_of_integer = float(integer / 2)

File: Week1/Day2/ExercisesXP/script.py (hunk 9 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:64]: Yes:
username = username.capitalize()

File: Week1/Day2/ExercisesXP/script.py (hunk 11 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:81]: Yes:
while pizza_topping != 'quit'

File: Week1/Day2/ExercisesXP/script.py (hunk 12 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:95]: while age != 'quit'

File: Week1/Day2/ExercisesXP/script.py (hunk 8 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:57]: Unfortunately, I don't understand exactly what you mean.

File: Week1/Day2/ExercisesXP/script.py (hunk 14 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:114]: Yes.
maybe:
list_allowed = [name for name in names if name not in names_for_removal]
print(list_allowed)

File: Week1/Day2/ExercisesXP/script.py (hunk 17 above)
Comment (David-7805) [reply Week1/Day2/ExercisesXP/script.py:136]: Maybe to print inside the while loop that came before:

print(f'I made you a {finished_sandwich}')

**Pull request merged**